user/caribou/module/src/DSO9254AEventConverter.cc

```diff
@@ -0,0 +1,377 @@
+	int16_t wfi = 0;
+	for(int i=block_posit + 3 + pream_words + (s+0)*chann_words/sgmnt_count;
+	        i<block_posit + 3 + pream_words + (s+1)*chann_words/sgmnt_count;
+	      //i<block_posit + 3 + pream_words+chann_words;
+	        i++){
+	  
+	  // coppy channel data from whole data block
+	  memcpy(&words.front(), &rawdata[i], points_per_words*sizeof(int16_t));
+	  
+	  for( auto word : words ){
+	    
+	    // fill vectors with time bins and waveform
+	    if( nch == 0 && s == 0 ){ // this we need only once
+	      time.push_back( wfi * dx.at(nch) + x0.at(nch) );
+	    }
+	    wave.push_back( (word * dy.at(nch) + y0.at(nch)) );
+	    
+	    
+	    // fill histogram
+	    hist->SetBinContent(  hist->FindBin( wfi * dx.at(nch) + x0.at(nch) ),
+	    			  (double)word * dy.at(nch) + y0.at(nch) );
+	    
+	    wfi++;
+	    
+	  } // words
+	  
+	} // waveform
+
+	
+	// store waveform vector and create vector entries for pedestal and amplitude
+	waves.push_back( wave );
+	ped.push_back( 0. );
+	amp.push_back( 0. );
+
+	// write and delete
+	if( generateRoot ){
+	  hist->Write();
+	}
+	delete hist;
+
+	
+      } // segments
+
+
+      // not only for each semgent but for each channel
+      wavess.push_back( waves );
+      peds.push_back( ped );
+      amps.push_back( amp );
+	
+
+      // update position for next iteration
+      block_posit += block_words+1;
+    } // for channel
+    
+  } // if numblock
+
+
+  // process waveform data
+  if( time.size() > 0 ){ // only if there are waveform data
+  
+
+    // re-iterate waveforms
+    // wavess.at(channel).at(segment).at(point)
+    // ped   .at(channel).at(segment)
+    for( int s = 0; s<peds.at(0).size(); s++ ){
+      for( int p = 0; p<time.size(); p++ ) {      	
+	
+	//try{ // hotfix in case off incomplete waveforms
+	  
+	// calculate pedestal
+	if( time.at(p) >= pedStartTime && time.at(p) < pedEndTime ){
+	  for( int c = 0; c<peds.size(); c++ ) 
+	    peds.at(c).at(s) += wavess.at(c).at(s).at(p);
+	}
+	
+	// get maximum amplitude in range
+	if( time.at(p) >= ampStartTime && time.at(p) < ampEndTime ){
+	  for( int c = 0; c<peds.size(); c++ ){ 
+	    if( amps.at(c).at(s) < wavess.at(c).at(s).at(p) - peds.at(c).at(s) )
+	      amps.at(c).at(s) = wavess.at(c).at(s).at(p) - peds.at(c).at(s);
+	  }
+	}
+
+	//}catch(...){} 
+	
+      } // points
+      
+      
+      // check:
+      /*
+      std::cout << std::endl;
+      for( int c = 0; c<peds.size(); c++ ){
+	std::cout << "  CHECK: channel " << c << " segment " << s << std::endl 
+	          << "         points  " << wavess.at(c).at(s).size() << std::endl
+		  << "         pedestl " << peds.at(c).at(s) << std::endl
+		  << "         ampli   " << amps.at(c).at(s) << std::endl
+		  << "         timestp " << timestamp << std::endl;
+      }
+      */
+  
+    } // segments
+    
+    
+    // Create a StandardPlane representing one sensor plane
+    // FIXME For now pushing just the first segment... need to decide how to treat several of them
```

Comment:
That's a good question - normally we have these separated in subevents - but at the stare of RawEvents from eudaq, not after decoding... :thinking:

Reply:
https://gitlab.cern.ch/corryvreckan/corryvreckan/-/merge_requests/489